src/layers/domain/repository/cpm_product_repository/v3.py

```diff
@@ -42,8 +141,7 @@ def read(self, product_team_id, product_id) -> CpmProduct:
         result = self.client.query(**args)
         items = [unmarshall(i) for i in result["Items"]]
         if len(items) == 0:
-            raise ItemNotFound(key=product_id)
-
+            raise ItemNotFound(key=f"{product_team_id}:{product_id}")
```

Comment:
Do we want the key to be made up of both product team id and product id here? By this stage it should have already been verified that the product team exists? 
(And this isn't the pk for products)

Reply:
Agreed this is more helpful - also tailor the error message to specify which object wasn't found, e.g. product team or product